Sources/Diagnostics/DiagnosticsEvent.swift

```diff
@@ -13,59 +13,84 @@
 
 import Foundation
 
-struct DiagnosticsEvent: Codable {
+/// When sending this to the backend `JSONEncoder.KeyEncodingStrategy.convertToSnakeCase` is used.
+struct DiagnosticsEvent: Codable, Equatable {
 
-    let version: Int = 1
-    let eventType: DiagnosticsEvent.EventType
-    let properties: [DiagnosticsPropertyKey: AnyEncodable]
+    private(set) var version: Int = 1
+    let name: EventName
+    let properties: Properties
     let timestamp: Date
 
-    enum CodingKeys: String, CodingKey {
-        case version, properties, timestamp, eventType
+    enum EventName: String, Codable, Equatable {
+        case httpRequestPerformed = "http_request_performed"
+        case appleProductsRequest = "apple_products_request"
+        case customerInfoVerificationResult = "customer_info_verification_result"
+        case maxEventsStoredLimitReached = "max_events_stored_limit_reached"
+        case applePurchaseAttempt = "apple_purchase_attempt"
     }
 
-}
-
-extension DiagnosticsEvent {
-
-    enum EventType: String, Codable {
-
-        case httpRequestPerformed
-        case appleProductsRequest
-        case customerInfoVerificationResult
-        case maxEventsStoredLimitReached
-        case applePurchaseAttempt
-
+    enum PurchaseResult: String, Codable, Equatable {
+        case verified
+        case unverified
+        case userCancelled = "user_cancelled"
+        case pending
     }
 
-    enum DiagnosticsPropertyKey: String, Codable {
+    struct Properties: Codable, Equatable {
+        let verificationResult: String?
+        let endpointName: String?
+        let responseTimeMillis: Int?
+        let storeKitVersion: String?
```

Comment:
I'm wondering if we should send the storeKitVersion for all events... but I think it's ok to keep it like this for now. That can be a future change if needed.

Reply:
Does it makes sense in all events? E.g. `get_offerings_started` or `get_customer_info_started`.